v4l2/v4l2_decode.cpp

```diff
@@ -857,6 +862,12 @@ void V4l2Decoder::getInputJob()
         if (m_output->isAllocationDone()) {
             m_state = kWaitSurface;
             DEBUG("early out, no surface");
+            if (m_isThumbnailMode) {
```

Comment:
Why we need this?  In thubmail mode, we will decode the first frame then get output. then it will decode the second frame, and get YAMI_DECODE_NO_SURFACE.  So we do not need do special thing for thumbnail mode here

Reply:
Yes, the first frame is decoded and output. but we still do not have chance to take it to v4l2, and not have chance to setDeviceEvent() to notify app either. If it enters kWaitSurface, it will block here.
but yes, consumeInput() is not mandatory. @xuguangxin